### PR TITLE
Async download performance

### PR DESCRIPTION
**Pull request by sblack-usu**

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4088](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4088)

**Review by hyi — APPROVED** [review]

File: hs_core/task_utils.py [hunk 1]
```diff
@@ -101,34 +81,16 @@ def get_or_create_task_notification(task_id, status='progress', name='', payload
...
             obj.save()
 
-        if obj.status == "progress":
-            if not task_exists(task_id):
-                obj.status = 'failed'
-                obj.save()
-
         return {
             'id': task_id,
+            'task_id': task_id,
```
Comment (hyi) [inline hs_core/task_utils.py:86]: why need to add another key task_id with the same value as id?

File: hs_core/task_utils.py [hunk 2]
```diff
@@ -101,34 +81,16 @@ def get_or_create_task_notification(task_id, status='progress', name='', payload
...
             'name': name,
             'status': obj.status,
-            'payload': obj.payload
+            'payload': obj.payload,
+            'download_path': obj.payload
```
Comment (hyi) [inline hs_core/task_utils.py:90]: why add download_path key here with the same value as payload?

**Review by hyi — COMMENTED** [review]
@sblack-usu Have left a couple of comments.

File: hs_core/task_utils.py (hunk 1 above)
Comment (sblack-usu) [reply hs_core/task_utils.py]: I mentioned it slack to you, but it does warrant a comment in the code.  It's for backwards compatibility with rest calls.

**Review by hyi — DISMISSED** [review]

**Review by hyi — APPROVED** [review]

**Pull request merged**